File: Subprogramas/Polimorfismo/Polimorfismo.php
<?php
//JavaScript
// Não suporta sobrecarga de subprogramas diretamente (o último método definido sobrescreve o anterior).
// Uma forma de simular subprogramas polimorficos é um único subprograma que filtra o dado.

class Polimorfismo {
    public function imprimir($value) {
        if (is_string($value)) {
            echo $value;
        } elseif (is_int($value)) {
            echo $value;
        }
    }
}

$p = new Polimorfismo();
$p->imprimir("Hello");  // imprime a string
$p->imprimir(10);       // imprime o inteiro
?>

File: Subprogramas/Polimorfismo/Polimorfismo.rs
//Rust não suporta diretamente sobrecarga de subprogramas. 
//O polimorfismo é implementado por meio de traits (semelhantes às interfaces em outras linguagens).

struct Polimorfismo;

impl Polimorfismo {
    fn print_str(&self, s: &str) {
        println!("{}", s);
    }

    fn print_int(&self, i: i32) {
        println!("{}", i);
    }
}

fn main() {
    let p = Polimorfismo;
    p.print_str("Hello");  // chama o método que aceita string
    p.print_int(10);       // chama o método que aceita int
}
